cos_registration_server/api/views.py

```diff
@@ -259,3 +266,110 @@ def delete(self, request: Request, uid: str) -> Response:
         dashboard = self._get_dashboard(uid)
         dashboard.delete()
         return Response(status=204)
+
+
+class PrometheusAlertRulesView(APIView):
+    """PrometheusAlertRules API view."""
+
+    def get(self, request: Request) -> Response:
+        """Prometheus Alert Rules get view.
+
+        request: Http GET request.
+        return: Http JSON response.
+        """
+        alert_rules = PrometheusAlertRule.objects.all()
+        rendered_rules = []
+
+        template_alert_rules = PrometheusAlertRule.objects.filter(
+            template=True
+        )
+        devices = Device.objects.all()
+
+        for device in devices:
+            for rule in device.prometheus_alert_rules.all():
+                if rule in template_alert_rules:
+                    env = Environment(
+                        variable_start_string="%%",
+                        variable_end_string="%%",
+                    )
+                    rule_string = yaml.dump(rule.rules)
+                    template = env.from_string(rule_string)
+                    context = {"juju_device_uuid": f"{device.uid}"}
+                    rendered_rule = template.render(context)
+                    rendered_rules.append(
+                        {
+                            "uid": rule.uid + "_" + device.uid,
```

Comment:
```suggestion
                            "uid": rule.uid + "/" + device.uid,
```

Reply:
changed in 088a0785869d2e23d8ddfdf7f31589f8bc4327f4